### PR TITLE
fix: Add separate type files for every folder

### LINKED ISSUE #10: Feature request: Improving Code Quality and Correcting Types in ReadmeEase Project
This issue is related to improving the code quality and correcting the types in the `ReadmeEase` project, which is written in TypeScript. During the development process, there may be instances where the types of the code are not correct or suitable, leading to potential issues and bugs in the application. To ensure that the project is robust and reliable, it is essential to fix these types and improve the code quality.

The objective of this issue is to identify and fix any incorrect or unsuitable types in the codebase and make changes as necessary to ensure that the code adheres to best practices and standards. This issue will involve a thorough review of the codebase, identifying areas where types are incorrect, and proposing and implementing changes to fix these issues.

The implementation of this issue will not only improve the code quality of the project but also ensure that the application is more reliable and efficient. It will also help to prevent potential bugs and errors, making it easier to maintain and extend the project in the future.

### PR DESCRIPTION
**Pull request by ehsaanchanga**
## Description

Added separate type files for each folder to improve code quality and correct types.

Fixes #10 

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please describe):

## Validation

- [ ] I have tested my changes locally and verified that they work as expected.
- [ ] I have updated the documentation (if applicable) to reflect the changes I've made.
- [ ] I have added unit tests (if applicable) to ensure my changes don't break existing functionality.

## Checklist

- [ ] I have read the [contributing guidelines](https://github.com/JunDevHarsh/ReadmeEase#contributing) and my pull request follows them.
- [ ] My code follows the code style of this project.
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added or updated tests to cover my changes.
- [ ] All new and existing tests passed.

## Labels

labels:

### CODE REVIEW

**Comment by JunDevHarsh** [comment]
@ehsaanchanga, Thank you for making a PR. There is an error that needs to be fixed. Here is the screenshot of it.

Screenshot 2023-02-24 172714

File: src/components/Modal/types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export type DeleteModalPropsType = {
+  id: string;
```
Comment (JunDevHarsh) [inline src/components/Modal/types.ts:2]: `id: string;` should be `id: string | string[]`

**Review by JunDevHarsh — CHANGES_REQUESTED** [review]
Need to make a little change in the file in `./src/components/Modal/Delete.tsx`

`id: string` should be `id: string | string[]`

**Comment by JunDevHarsh** [comment]
> @JunDevHarsh fixed the types file

@ehsaanchanga, you have to update the `DeleteModalPropsType` block, not the `FileItemType` block. Here is the code you can look at:

```javascript
export type DeleteModalPropsType = {
  id: string; // change this line to 'id: string | string[]'
  toggleModal: React.Dispatch<React.SetStateAction<boolean>>;
};
```

**Comment by ehsaanchanga** [comment]
@JunDevHarsh sorry for that . its done now

**Review by JunDevHarsh — APPROVED** [review]

**Comment by JunDevHarsh** [comment]
Thank you @ehsaanchanga for your contribution,

**Pull request merged**